atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheMetrics.java

```diff
@@ -122,4 +120,24 @@ public void setTransactionCacheInstanceCountGauge(Gauge<Integer> getCacheMapCoun
         metricsManager.registerOrGetGauge(
                 CacheMetrics.class, AtlasDbMetricNames.LW_TRANSACTION_CACHE_INSTANCE_COUNT, () -> getCacheMapCount);
     }
+
+    public void setEventsHeldInMemory(Gauge<Integer> eventsGauge) {
+        metricsManager.registerOrGetGauge(
```

Comment:
This is not new, but register or get is quite sketchy to use. Maybe we do it so that tests don't fail, but if we actually ever get instead of register, that's quite scary